### ISSUE TITLE
Check the dataverse and dataset pages for memory use and/or leaks

### ISSUE THREAD

**Issue by landreev** [IQSS/dataverse#7803]
A sub-issue of #7788, the overall performance improvement issue. 
We have automated scripts for spinning up builds, running load test on "sample data" and generating memory use/garbage collection plots. (We used the technique to identify the fact that our anonymous sessions were essentially one big memory leak - since we used to allow them to live for many hours, and none of the associated objects could be garbage-collected until the EOL of the session).
Let's build on that and test the dataverse (and dataset?) page(s), see what the memory use patterns are there now and whether there are any memory leaks we need to worry about.

**Comment by djbrooke** [comment]
- We should re-run the scripts that we added for the memory leak detection, and adjust based on the result

**Comment by madryk** [comment]
If you're using S3 as storage driver there could be some issue with not closed streams (s3 is very fragile in such cases - it could easily block rendering of dataverse.xhtml or dataset.xhtml - depends if there are some thumbnail to render or not).
I don't know if this the case but in `DownloadInstanceWriter` if we're past the code:
https://github.com/IQSS/dataverse/blob/a875db778a53db3353e6b3095e52bdbe6fff01e9/src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java#L388-L389
and the client won't bother to download the requested file, then there will be an exception somewhere in the lines:
https://github.com/IQSS/dataverse/blob/a875db778a53db3353e6b3095e52bdbe6fff01e9/src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java#L436-L445
as a result there would be no invocation of code:
https://github.com/IQSS/dataverse/blob/a875db778a53db3353e6b3095e52bdbe6fff01e9/src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java#L470
As a result s3 stream will be never closed.
Repeat 256 times:
https://github.com/IQSS/dataverse/blob/a875db778a53db3353e6b3095e52bdbe6fff01e9/src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java#L1110-L1118
And all downloads are blocked

**Comment by qqmyers** [comment]
Good catch! https://github.com/IQSS/dataverse/blob/a875db778a53db3353e6b3095e52bdbe6fff01e9/src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java#L388-L389 should get wrapped in a try() { so that the stream is closed.

**Comment by cmbz** [comment]
2025-01-22: Closing as not planned (as part of our backlog reduction effort).
